matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/store/db/RealmCryptoStore.kt

```diff
@@ -756,6 +759,44 @@ internal class RealmCryptoStore @Inject constructor(
         return inboundGroupSessionToRelease[key]
     }
 
+    override fun getCurrentOutboundGroupSessionForRoom(roomId: String): OutboundGroupSessionWrapper? {
+        return doWithRealm(realmConfiguration) { realm ->
+            realm.where<CryptoRoomEntity>()
+                    .equalTo(CryptoRoomEntityFields.ROOM_ID, roomId)
+                    .findFirst()?.outboundSessionInfo?.let { entity ->
+                        entity.getOutboundGroupSession()?.let {
+                            OutboundGroupSessionWrapper(
+                                    it,
+                                    entity.creationTime ?: 0
+                            )
+                        }
+                    }
+        }
+    }
+
+    override fun storeCurrentOutboundGroupSessionForRoom(roomId: String, outboundGroupSession: OlmOutboundGroupSession?) {
+        // we can do this async, as it's just for restoring on next launch
+        // the olmdevice is caching the active instance
+        // this is called for each sent message (so not high frequency), thus we can use basic realm async without
+        // risk of reaching max async operation limit?
+        doRealmTransactionAsync(realmConfiguration) { realm ->
+            realm.where<CryptoRoomEntity>()
+                    .equalTo(CryptoRoomEntityFields.ROOM_ID, roomId)
+                    .findFirst()?.let { entity ->
```

Comment:
You can use `CryptoRoomEntity.getById()`

Reply:
Done